### PR TITLE
 MISUV-397: Create view for deductions

### PR DESCRIPTION
**Pull request by scullythornton1**

### CODE REVIEW

File: conf/messages [hunk 1]
```diff
@@ -71,6 +71,15 @@ income_breakdown.table.bbsi                                     = Interest from
 income_breakdown.table.dividends                                = Dividends from UK companies
 income_breakdown.total                                          = Total income
 
+## Deduction Breakdown Page ##
+deduction_breakdown.heading                                        = Deductions
+deduction_breakdown.dates                                          = 6 April {0} to 5 April {1}
+deduction_breakdown.guidance_software                              = This is a summary of your income for the tax year {0}-{1}. You can see more details in your
+deduction_breakdown.guidance_software.link                         = record-keeping software
+deduction_breakdown.table.personal_allowance                       = Personal Allowance
+deduction_breakdown.table.gift_of_investments_and_property_to_charity  = Gift of investments and property to charity
```
Comment (JSimmons072019) [inline conf/messages:80]: Re- align the equals signs so that they all line up for this section

File: conf/messages.cy [hunk 2]
```diff
@@ -71,6 +71,15 @@ income_breakdown.table.bbsi                                     = Llog o fanciau
...
 income_breakdown.total                                          = Cyfanswm yr incwm
 
+## Deduction Breakdown Page ##
+deduction_breakdown.heading                                        = Didyniadau
+deduction_breakdown.dates                                          = 6 Ebrill {0} hyd at 5 Ebrill {1}
+deduction_breakdown.guidance_software                              = Dyma grynodeb o’ch incwm ar gyfer blwyddyn dreth {0}-{1}. Gallwch weld rhagor o fanylion yn eich
+deduction_breakdown.guidance_software.link                         = meddalwedd cadw cofnodion
+deduction_breakdown.table.personal_allowance                       = Lwfans Personol
+deduction_breakdown.table.gift_of_investments_and_property_to_charity  = Rhoddion ar ffurf buddsoddiadau ac eiddo i elusennau
+deduction_breakdown.total                                          = Cyfanswm y didyniadau
```
Comment (JSimmons072019) [inline conf/messages.cy:81]: Re- align the equals signs so that they all line up for this section

File: app/views/deductionBreakdown.scala.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+   @messages("deduction_breakdown.guidance_software", s"${taxYear-1}", s"$taxYear")
+   <a href="https://www.gov.uk/guidance/find-software-thats-compatible-with-making-tax-digital-for-income-tax" target="_blank">@messages("deduction_breakdown.guidance_software.link") <span class="visually-hidden">@messages("obligations.opens-in-new-tab")</span></a>.
+  </p>
+
+  <table>
+   @tableRow("personal_allowance", calcModel.calcDataModel.allowancesAndDeductions.personalAllowance)
+   @tableRow("gift_of_investments_and_property_to_charity", calcModel.calcDataModel.allowancesAndDeductions.giftOfInvestmentsAndPropertyToCharity)
+   <tr class="no-border-bottom">
+    <td class="total-section">@messages("deduction_breakdown.total")</td>
+    <td class="numeric total-section">@calcModel.calcDataModel.allowancesAndDeductions.totalAllowancesAndDeductions.getOrElse(BigDecimal(0)).toCurrency</td>
```
Comment (AlexRimmerHMRC) [inline app/views/deductionBreakdown.scala.html:53]: The value used on the overview page is calculation.allowancesAndDeductions.totalAllowancesDeductionsReliefs from the model, the value used on the overview and the value you have used might not match up.

File: app/views/deductionBreakdown.scala.html [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+@(calcModel: CalcDisplayModel, taxYear: Int)(implicit request: Request[_], messages: Messages, appConfig: config.FrontendAppConfig, user: auth.MtdItUser[_])
+
+@tableRow(messageCode: String, dataItem: Option[BigDecimal]) = {
+  @if(dataItem.isDefined) {
+   <tr id="@messageCode">
+    <td>@messages(s"deduction_breakdown.table.$messageCode")</td>
+    <td class="numeric">@{dataItem.get.toCurrency}</td>
+   </tr>
+  }
```
Comment (AlexRimmerHMRC) [inline app/views/deductionBreakdown.scala.html:30]: can be simplified, doesn't need the .isDefined and .get, can use dataItem.map { data => data is not optional here anymore }

**Pull request merged**